packages/core/src/generator/index.ts

```diff
@@ -17,6 +17,28 @@ export class UnoGenerator {
     this.config = resolveConfig(userConfig, defaults)
   }
 
+  resolveBreakpoints() {
+    let breakpoints: Record<string, string> | undefined
+    if (this.userConfig && this.userConfig.theme)
+      breakpoints = (this.userConfig.theme as any).breakpoints
+
+    if (!breakpoints && this.config.theme)
+      breakpoints = (this.config.theme as any).breakpoints
+
+    return breakpoints
+  }
+
+  resolveVerticalBreakpoints() {
```

Comment:
Those two should belong to preset-mini instead of core. The core should be agnostic to how the theme object has been defined.